File: ejercicio1.php
<?php
// Uno esta sentado frente al espejo y mira la imagen de un reloj situado detras de usted. 
// Quiere saber que hora es. El reloj es un reloj tradicional que marca 12 horas tiene un minutero y uno que marca a hora. 
// Dado una cadena de numero que indica como se ve en el espejo. Se le pide devolver una cadena de nuemeros con la hora real.
// Por ejemplo si nos da la cadena 10:00 La respuesta será 02:00. 
// 01:45 salida 10 y 45. 03:40 salida 09:20.

$hora = ["03:30"];

for ($i = 0; $i < count($hora); $i++) { // Iniciamos un bucle para recorrer todas las horas del array
    $horaEspejo = $hora[$i]; // Obtenemos la hora actual del array usando el índice
    // Separar hora y minutos
    $partes = explode(':', $horaEspejo); // Dividimos la cadena de texto usando ':' como separador
    $h = $partes[0]; // Tomamos la primera parte (antes de los dos puntos) como la hora
    $m = $partes[1]; // Tomamos la segunda parte (después de los dos puntos) como los minutos

    // Calcular hora real
    $hReal = 12 - $h; // Restamos la hora leída a 12 para obtener la hora real invertida
    if ($hReal == 0) { // Si el resultado de la resta es 0 (ej. 12 - 12)
        $hReal = 12; // Corregimos la hora real para que sea 12
    }

    // Calcular minutos reales
    $mReal = 60 - $m; // Restamos los minutos leídos a 60 para obtener los minutos reales
    if ($mReal == 60) { // Si el resultado es 60 (ej. 60 - 0 minutos)
        $mReal = 0; // Corregimos los minutos reales para que sean 0
    }

    // Agregar cero a la izquierda si hace falta
    if ($hReal < 10) { // Verificamos si la hora real tiene un solo dígito (es menor a 10)
        $hReal = "0" . $hReal; // Concatenamos un '0' al principio de la hora
    }
    if ($mReal < 10) { // Verificamos si los minutos reales tienen un solo dígito (es menor a 10)
        $mReal = "0" . $mReal; // Concatenamos un '0' al principio de los minutos
    }

    echo "Espejo: " . $horaEspejo . " -- Hora real: " . $hReal . ":" . $mReal . "<br>"; // Imprimimos la hora original y la calculada
}
?>

File: ejercicio1.py
hora = ["01:30"] # Definimos una lista con la hora espejo de ejemplo

for i in range(len(hora)): # Recorremos la lista de horas usando un índice
    horaEspejo = hora[i] # Obtenemos la hora actual del ciclo
    
    # Separar hora y minutos
    partes = horaEspejo.split(':') # Dividimos el texto por los dos puntos para separar hora y minutos
    h = int(partes[0]) # Convertimos la parte de la hora a entero
    m = int(partes[1]) # Convertimos la parte de los minutos a entero

    # Calcular hora real
    hReal = 12 - h # Restamos la hora espejo de 12 para obtener la hora real
    if hReal == 0: # Si la resta da 0 (es decir, eran las 12)
        hReal = 12 # Ajustamos la hora real a las 12

    # Calcular minutos reales
    mReal = 60 - m # Restamos los minutos espejo de 60 para obtener los minutos reales
    if mReal == 60: # Si la resta da 60 (es decir, eran las en punto :00)
        mReal = 0 # Ajustamos los minutos reales a 0

    # Agregar cero a la izquierda si hace falta
    hRealStr = str(hReal) # Convertimos la hora real calculada a texto
    if hReal < 10: # Si la hora es menor a 10 (un solo dígito)
        hRealStr = "0" + hRealStr # Le agregamos un "0" al principio

    mRealStr = str(mReal) # Convertimos los minutos reales calculados a texto
    if mReal < 10: # Si los minutos son menores a 10 (un solo dígito)
        mRealStr = "0" + mRealStr # Le agregamos un "0" al principio

    print(f"Espejo: {horaEspejo} -- Hora real: {hRealStr}:{mRealStr}") # Imprimimos el resultado formateado
